src/java.base/share/classes/com/sun/crypto/provider/GaloisCounterMode.java

```diff
@@ -494,40 +503,68 @@ int encrypt(byte[] in, int inOfs, int len) {
      *
      * @param in the buffer with the input data to be encrypted
      * @param inOfs the offset in <code>in</code>
-     * @param len the length of the input data
+     * @param inLen the length of the input data
      * @param out the buffer for the result
      * @param outOfs the offset in <code>out</code>
      * @return the number of bytes placed into the <code>out</code> buffer
      */
     int encrypt(byte[] in, int inOfs, int inLen, byte[] out, int outOfs) {
-        checkDataLength(processed, inLen);
+        checkDataLength(inLen, getBufferedLength());
+        ArrayUtil.nullAndBoundsCheck(in, inOfs, inLen);
 
```

Comment:
Yes, I'll add it back in.. I wanted to change it so it would verify correctly output length rather than inLen, which is not necessarily right given there is ibuffer available.  Maybe Ill try to do that in a later update.